harvester/src/test/scala/za/co/absa/spline/harvester/SparkLineageInitializerSpec.scala

```diff
@@ -94,108 +87,117 @@ class SparkLineageInitializerSpec extends FunSpec with BeforeAndAfterEach with M
+        splineConfiguration getString keyDefinedInJVMAndSpline shouldEqual valueFromJVM
+        splineConfiguration getString keyDefinedInHadoopAndSpline shouldEqual valueFromHadoop
+        splineConfiguration getString keyDefinedInHadoopAndSpark shouldEqual valueFromHadoop
+        splineConfiguration getString keyDefinedInSparkAndSpline shouldEqual valueFromSpark
+        splineConfiguration getString keyDefinedInSplineOnly shouldEqual valueFromSpline
+        splineConfiguration getString "key.undefined" shouldBe null
+      })
     }
   }
 
   describe("enableLineageTracking()") {
     it("should not allow double initialization") {
       intercept[IllegalStateException] {
-        val session = createSession
-        session
-          .enableLineageTracking(createConfigurer(session)) // 1st is fine
-          .enableLineageTracking(createConfigurer(session)) // 2nd should fail
+        withNewSession(session =>
+          session
+            .enableLineageTracking(createConfigurer(session)) // 1st is fine
+            .enableLineageTracking(createConfigurer(session)) // 2nd should fail
+        )
       }
     }
 
     it("should return the spark session back to the caller") {
-      val session = createSession
-      session.enableLineageTracking(createConfigurer(session)) shouldBe session
+      withNewSession(session =>
+        session.enableLineageTracking(createConfigurer(session)) shouldBe session
+      )
     }
 
     describe("modes") {
 
       it("should disable Spline and proceed, when is in BEST_EFFORT (default) mode") {
-        configuration.setProperty(MODE, BEST_EFFORT.toString)
-        val sparkSession = createSession
-        sparkSession.enableLineageTracking(createFailingConfigurer(sparkSession))
-        assertSplineIsDisabled()
+        withNewSession(sparkSession => {
+          configuration.setProperty(MODE, BEST_EFFORT.toString)
+          sparkSession.enableLineageTracking(createFailingConfigurer(sparkSession))
+          assertSplineIsDisabled()
+        })
       }
 
       it("should disable Spline and proceed, when is in DEFAULT mode") {
-        configuration.clearProperty(MODE) // default mode is BEST_EFFORT
-        val sparkSession = createSession
-        sparkSession.enableLineageTracking(createFailingConfigurer(sparkSession))
-        assertSplineIsDisabled()
+        withNewSession(sparkSession => {
+          configuration.clearProperty(MODE) // default mode is BEST_EFFORT
+          sparkSession.enableLineageTracking(createFailingConfigurer(sparkSession))
+          assertSplineIsDisabled()
+        })
       }
 
       it("should abort application, when is in REQUIRED mode") {
         intercept[Exception] {
-          configuration.setProperty(MODE, REQUIRED.toString)
-          val session = createSession
-          session.enableLineageTracking(createFailingConfigurer(session))
+          withNewSession(session => {
```

Comment:
why not use something like @before that is used in JUnit

Reply:
those are alternative techniques. In fact in scalatests you can choose between three different techniques for implementing fixtures. The one that uses 'before' and 'after' methods is useful when you want to have more fine-grained control on the fixture init and release steps. And it also aborts the entire suite in case of an error during a fixture initialization. So before-and-after is a more heavy-weight technique.
The one that you pointer (`withNewsession`-method) is called a _loan-pattern_. It allows you to decide on the fixture details on a per test method basis. It is more suitable for tests when each methods might need different state of a set of fixtures.
Basically `SparkFixture` implements both patterns and can be used in both ways.